apps/vaporgui/SettingsEventRouter.cpp

```diff
@@ -197,43 +198,110 @@ void SettingsEventRouter::_changesPerSaveChanged() {
 void SettingsEventRouter::_chooseAutoSaveFile() {
 	SettingsParams* sParams = (SettingsParams *) GetActiveParams();
 
-    QString fileName = QFileDialog::getSaveFileName(
-        _autoSaveFileButton, 
-        tr("Select auso-save VAPOR session file"),
+	QString fileName = QFileDialog::getSaveFileName(
+		_autoSaveFileButton, 
+		tr("Select auso-save VAPOR session file"),
 		QString::fromStdString( sParams->GetAutoSaveSessionFile() ),
-        tr("Vapor 3 Session Files (*.vs3)")
-    );
+		tr("Vapor 3 Session Files (*.vs3)")
+	);
 
-	if (! fileName.isEmpty()) 
-    {
+	bool goodToWrite = FileOperationChecker::FileGoodToWrite(fileName); 
+	if (! fileName.isEmpty() && goodToWrite) {
```

Comment:
Testing both if `fileName` is empty and if it's good to write is redundant. That is, if a fileName were empty, it won't pass the "GoodToWrite" test.